### PR TITLE
#55 - Generic Modal

### PR DESCRIPTION
**Pull request by chanchal-sadhukhan**
## Description
_A few sentences describing the overall goals of the pull request's commits._

## Screenshot
Screencast from 12-22-2021 10_47_27 PM

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._
- _Description of manual test 1_
- _Description of manual test 2_
- _Description of manual test 3_

## Platform tested
- [ ] Large
- [ ] Tablet
- [ ] Mobile

### CODE REVIEW

File: src/components/primitive/generi-cmodal/generic-modal.component.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  font-size: 2rem;
+  font-weight: bold;
+  color: ${(props) => props.theme.colors.textLight};
+  z-index: 999;
+
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
```
Comment (Koushith) [inline src/components/primitive/generi-cmodal/generic-modal.component.tsx]: @cksadhukhan can you please move this to separate styles file?

File: src/components/primitive/generi-cmodal/generic-modal.component.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        afterOpen={afterOpen}
+        beforeClose={beforeClose}
+        onBackgroundClick={toggleModal}
+        onEscapeKeydown={toggleModal}
+        backgroundProps={{ opacity }}
+      >
+        <CloseModal onClick={toggleModal}>X</CloseModal>
+        <Box vCenter hCenter paddingTop={2}>
+          <Text bold {...title} />
+        </Box>
```
Comment (Koushith) [inline src/components/primitive/generi-cmodal/generic-modal.component.tsx]: Title should be optional as this is a generic modal which can be reused along with loaders.

File: src/components/primitive/generi-cmodal/generic-modal.component.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            label={{ text: "Close", color: "black" }}
+            variant="ghost"
+            onClick={onClose}
+          />
+          <ModalButton
+            label={{ text: "Continue" }}
+            variant="small"
+            onClick={onClose}
+          />
+        </Box>
```
Comment (Koushith) [inline src/components/primitive/generi-cmodal/generic-modal.component.tsx]: Even this should be optional.

File: src/components/primitive/generi-cmodal/generic-modal.component.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+import styled from "styled-components";
+import Modal, { ModalProvider, BaseModalBackground } from "styled-react-modal";
+import { GenericModalComponentProps } from "./generic-modal.component.props";
+import { Box, Text, Button } from "..";
+
+const FadingBackground = styled(BaseModalBackground)`
+  opacity: ${(props) => props.opacity};
```
Comment (Koushith) [inline src/components/primitive/generi-cmodal/generic-modal.component.tsx]: I'm getting "opacity is not defined" while using this component.

**Review by Koushith — CHANGES_REQUESTED** [review]

**Review by koshikraj — APPROVED** [review]

**Review by Koushith — APPROVED** [review]

**Comment by Koushith** [comment]
LGTM

**Pull request merged**